app/models/notification.rb

```diff
@@ -79,7 +84,36 @@ def unarchive_if_updated
   def update_from_api_response(api_response, unarchive: false)
     attrs = Notification.attributes_from_api_response(api_response)
     self.attributes = attrs
+    update_subject
```

Comment:
Although your front end logic is behind the feature flag, this isn't and should be(?)

Though we should do it in conditionally in `update_subject` itself, and fill it what subject stuff we can without needing the permissions for full subject syncing (Always create a subject and fill in with info we currently fetch).